src/commands/admin_commands.py

```diff
@@ -0,0 +1,128 @@
+import os
+import logging
+from discord import app_commands, Interaction
+
+# Retrieve the list of admin usernames from the .env file
+ADMIN_USERS = os.getenv("ADMIN_USERS", "").split(",")
+
+
+class AdminCommands(app_commands.Group):
+    def __init__(self, gemini_bot):
+        super().__init__(name="admin", description="Admin commands for DuckBot setup.")
+
+        # Add subgroups to the main admin group
+        self.set = SetSubGroup(self.check_admin)
+        self.reset = ResetSubGroup(self.check_admin, gemini_bot)
+
+        # Register subgroups
+        self.add_command(self.set)
+        self.add_command(self.reset)
+
+    async def check_admin(self, interaction: Interaction) -> bool:
+        user_name = interaction.user.name
+        logging.info(f"Checking admin status for user: {user_name}")
+
+        if user_name in ADMIN_USERS:
+            logging.info(f"User {user_name} is authorized.")
+            return True
+        else:
+            logging.warning(f"User {user_name} is not authorized.")
+            return False
+
+    async def interaction_check(self, interaction: Interaction) -> bool:
+        """Restrict all admin commands visibility to authorized users only."""
+        is_admin = await self.check_admin(interaction)
+        if is_admin:
+            return True
+        await interaction.response.send_message("Unauthorized", ephemeral=True)
+        return False
+
+    @app_commands.command(
+        name="log-info", description="Display all current environment variables."
+    )
+    async def log_info(self, interaction: Interaction):
+        """Command to log and display all relevant environment variables."""
+        if not await self.check_admin(interaction):
+            await interaction.response.send_message("Unauthorized", ephemeral=True)
+            return
+
+        # Collect environment variable values
+        guild_id = os.getenv("GUILD_ID", "Not Set")
+        skullboard_channel_id = os.getenv("SKULLBOARD_CHANNEL_ID", "Not Set")
+        required_reactions = os.getenv("REQUIRED_REACTIONS", "Not Set")
+        tenor_api_key = os.getenv("TENOR_API_KEY", "Not Set")
+        gemini_api_key = os.getenv("GEMINI_API_KEY", "Not Set")
```

Comment:
You can remove the api key ones since it's not that useful unlike the other variables.